garage/tf/samplers/off_policy_vectorized_sampler.py

```diff
@@ -118,20 +127,20 @@ def obtain_samples(self, itr, batch_size):
                                                    env_infos, dones):
                 if running_paths[idx] is None:
                     running_paths[idx] = dict(
-                        rewards=[],
-                        env_infos=[],
-                    )
+                        rewards=[], env_infos=[], dones=[])
                 running_paths[idx]["rewards"].append(reward)
                 running_paths[idx]["env_infos"].append(env_info)
+                running_paths[idx]["dones"].append(done)
 
-                if done:
+                if done or n_samples >= batch_size:
```

Comment:
This would generate a `paths` which have an undone path right? So we would have an `undiscounted_returns` like `[98, 2]` (`max_path_length = 100` and `env.reward = 1`)?

Reply:
Yes. `paths` may contain undone paths. However, each path is one element in `paths`. In your example, the `undiscounted_returns` will be like `[ 'undiscounted_returns': [1], 'undiscounted_returns': [1]]`. The first `[1]` is the end of a path. The second `[1]` is the first sample of a path. Because `process_sample()` doesn't have access to previous steps, `return` cannot be correctly computed.